cmd/btcctl/httpclient.go

```diff
@@ -0,0 +1,132 @@
+package main
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+
+	"github.com/btcsuite/btcd/btcjson/v2/btcjson"
+	"github.com/btcsuite/go-socks/socks"
+)
+
+// newHTTPClient returns a new HTTP client that is configured according to the
+// proxy and TLS settings in the associated connection configuration.
+func newHTTPClient(cfg *config) (*http.Client, error) {
+	// Configure proxy if needed.
+	var dial func(network, addr string) (net.Conn, error)
+	//var proxyFunc func(*http.Request) (*url.URL, error)
+	if cfg.Proxy != "" {
+		proxy := &socks.Proxy{
+			Addr:     cfg.Proxy,
+			Username: cfg.ProxyUser,
+			Password: cfg.ProxyPass,
+		}
+		dial = func(network, addr string) (net.Conn, error) {
+			fmt.Println("Proxy dial called with", network, addr)
+			c, err := proxy.Dial(network, addr)
+			if err != nil {
+				fmt.Println("Proxy error", err)
+				return nil, err
+			}
+			fmt.Println("proxy dial complete")
+			return c, nil
+		}
+
+		// TODO(davec): Get the proxy working.
```

Comment:
Friendly reminder to address the `TODO`, and commented out sections below and above before merging. However, looking at the function assignment above, it looks as if you've already done this. `dial` gets initialized to a version which dials the `SOCKS5` proxy as an intermediary step.